### PR TITLE
Only benchmark on the latest Go version

### PR DESCRIPTION
**Pull request by rabbbit**
See  https://github.com/uber-go/ratelimit/issues/86#issuecomment-1146895304

Also:
- update to checkoutv3 (why not)
- use setup-go action to both:
  - pick up the lastest go version
  - do appropriate caching

### CODE REVIEW

File: .github/workflows/benchmark.yaml [hunk 1]
```diff
@@ -10,22 +10,10 @@ jobs:
     name: Benchmark
     runs-on: ubuntu-latest
     steps:
-      - name: Setup Go
-        uses: actions/setup-go@v2
+      - uses: actions/checkout@v3
+      - uses: actions/setup-go@v3
```
Comment (rabbbit) [inline .github/workflows/benchmark.yaml:14]: I'm switching order here based on https://github.com/actions/setup-go - they seem to recommend first, then setup-go.

I guess it makes sense given we want `setup-go` to do all the caching too.

**Comment by storozhukBM** [comment]
It failed after merge because it still runs on `go version go1.17.10 linux/amd64`

**Comment by storozhukBM** [comment]
Easy way to fix it would be to move
```
	@$(GOBIN)/benchstat stat.txt
	@$(GOBIN)/benchstat -csv stat.txt > stat.csv
	@$(GOBIN)/benchart 'RateLimiter;xAxisType=log' stat.csv stat.html
	@open stat.html
```

into a separate `make` target

**Comment by rabbbit** [comment]
Yeah, I was expecting the `latest` in the action to execute Go1.18. Somehow it didn't - I deferred debuging the action.

Separately make target sgtm too.

**Pull request merged**